partition/public.go

```diff
@@ -19,17 +19,13 @@
 
 package partition
 
-import (
-	"github.com/ubuntu-core/snappy/helpers"
-)
-
 // BootloaderDir returns the full path to the (mounted and writable)
 // bootloader-specific boot directory.
 func BootloaderDir() string {
-	if helpers.FileExists(bootloaderUbootDir()) {
-		return bootloaderUbootDir()
-	} else if helpers.FileExists(bootloaderGrubDir()) {
-		return bootloaderGrubDir()
+	if u := newUboot(); u != nil {
+		return u.Dir()
+	} else if g := newGrub(); g != nil {
+		return g.Dir()
 	}
 
 	return ""
```

Comment:
I've looked further, and found a single use of SetBootVar outside of `partition/`, within `snappy/kernel.go`. Why don't we just expose the bootloader interface as `Bootloader`, define a `FindBootloader` function returning `(Bootloader, error)`, and get rid of all of those wrappers (SetBootVar, GetBootVar, etc)?